app.py

```diff
@@ -86,20 +115,57 @@ def GitPull():
 
 @app.route('/', methods=['GET', 'POST'])
 def index():
-    global current
+    global current, queue
     form = ReservationForm()
     if form.validate_on_submit():
         name = form.name.data
-        minutes = form.minutes.data
         ipAddress = request.remote_addr
-        queue.append(Reservation(name, minutes, ipAddress))
+        dateTimeNow = floorDateTime(datetime.now())
+        minutes = timedelta(minutes = form.minutes.data)
+        # If there are no reservations
+        if current is None and len(queue) == 0:
+            startTime = dateTimeNow
+        # If there is a current reservation, but none in the queue
+        elif len(queue) == 0:
+            startTime = current.endTime
+        # If there is no current reservation, and the new reservation will end before the first next reservation starts
+        elif current is None and dateTimeNow + minutes <= queue[0].startTime:
+            startTime = dateTimeNow
+        # If there is no current reservation, and there is only one reservation in the queue
+        elif current is None and len(queue) == 1:
+            startTime = queue[0].endTime
+        # If the new reservation would not fit between the current and the first reservation
+        else:
+            currentItem = queue[0]
```

Comment:
Make current item `current` and start loop at 1. If the current has end time 12:12 and queue[0] starts at 12:20, a reservation of 5 minutes will now just pop to the end of the queue. By setting the currentItem to current and start the queue at 0 it should set the right start time in the while loop

Reply:
1e3cf13091c657d8b7dbfedd507644856cc27a14